sycl/doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc

```diff
@@ -0,0 +1,220 @@
+	
+```c++
+template <typename T, size_t Rows=std::dynamic_extent, size_t Cols=std::dynamic_extent, matrix_layout l = row_major>
+struct matrix;
+enum class matrix_layout {
+  row_major,
+  col_major,
+  packed_a,
+  packed_b
+};
+```
+### Matrix Operations
+We define three new functions needed to perform the main and common operations on matrices. This set of functions can be easily extended if the tensor hardware implements new features.
+The base pointer determines the starting address of the matrix to be loaded/stored.
+`layout` determines whether the data are being read/written in a row (`row_major`), column major (`column_major`) fashion, or if the data has already been transformed into VNNI format (`packed_a`, `packed_b`). 
+
+```c++
+void matrix_load(matrix<>A, T *base, unsigned stride, matrix_layout l = row_major);  
+```
+This function loads data from memory to the 2d tiles of AMX that is a 2d storage.
+```c++
+void matrix_store(matrix<>A, T *base, unsigned stride, matrix_layout l = row_major); 
+```
+This function stores the data from the 2d tiles back to memory.
+```c++
+matrix<> matrix_mad(matrix<>A, matrix<>B, matrix<>C);
+```
+
+The matrix multiply and add function performs the multiply operation on the matrices `A` and `B`, accumulate the result with `C` and return the result.
+
+## Integration with DPC++
+When using the matrix interface in a DPC++ kernel, additional semantics have to be added to define the memory scope of the matrix object and the execution scope of its operations. In the context of DPC++, `matrix` is distributed among an execution unit. In practice this can be one work-item, the work-items in a sub-group, or the work-items in a work-group.
+
+Since the matrix functions are group operations (as defined in Section 4.17.3 of the SYCL specification), the matrix API has to be accessed by all the work-items in the group in a non-diverged control flow. The `Group` template argument can be a work-group or a sub-group. These functions will be called once by all the work items in a group.
+
+### Memory Scope
+For the memory scope, we have two solutions. The long term solution is to use the proposed https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/LocalMemory/SYCL_INTEL_local_memory.asciidoc[`group_local_memory` extension] to allocate the matrix in local memory associated with a SYCL group as shown in the example below.
+
+```c++
+multi_ptr<matrix<T>, address_space::local_space> tA_ptr = group_local_memory<matrix<sub_group, int8_t, tM, tN>>(sg);
+```
+However, sub-group local memory is not yet well defined in DPC++. Moreover, the representation of this notion in LLVM IR and SPIR-V is not yet clear. Hence, for this proposal, we will proceed with adding the memory scope as an additional constructor argument as follows:
+
+```c++
+joint_matrix<sub_group, int8_t, tM, tN> tA(sg); 
```

Comment:
The rest of the comments have been addressed in the document.
But for these specific questions:
>>> Is the DPC++ type called matrix or joint_matrix? 
As long as we don't have  group_local_memory, joint_matrix will be used instead of matrix.
>>> What is sub_group here?
sub_group is sycl::sub_group

Do we really need both the sub_group template parameter and also the sg constructor parameter?
>>>Since the user has to specify the other template arguments: joint_matrix<int, 4, 4>, CTAD no longer works and C++ cannot deduce the group argument.